lib/plugin/instance.rb

```diff
@@ -1442,6 +1442,12 @@ def register_topic_preloader_associations(fields)
     DiscoursePluginRegistry.register_topic_preloader_association(fields, self)
   end
 
+  # When loading /categories with topics, preload topic associations
+  # using register_category_list_topics_preloader_associations(:association_name)
```

Comment:
I know not all of our APIs do this but I've found that the APIs that document the params and provide examples much easier to understand and use. 

Something like this: 

https://github.com/discourse/discourse/blob/9db89c20f7e64cd73cb116045b8b1840778a00d3/lib/plugin/instance.rb#L1283-L1313